### PR TITLE
Update to use the latest maven scala plugin

### PR DESCRIPTION
**Pull request by chtyim**
- It is needed in order to generate scaladoc jar.

### CODE REVIEW

File: cdap-examples/deploy_pom.xml [hunk 1]
```diff
@@ -65,64 +65,118 @@
...
+          <configuration>
+            <argLine>-Xmx2048m -Djava.awt.headless=true -XX:MaxPermSize=256m</argLine>
+            <redirectTestOutputToFile>${surefire.redirectTestOutputToFile}</redirectTestOutputToFile>
+            <reuseForks>false</reuseForks>
+            <reportFormat>plain</reportFormat>
+            <systemPropertyVariables>
+              <java.io.tmpdir>${project.build.directory}</java.io.tmpdir>
+            </systemPropertyVariables>
+            <includes>
+              <include>**/*TestsSuite.java</include>
```
Comment (awholegunch) [inline cdap-examples/deploy_pom.xml]: "TestsSuite"? Should this be generalized to cover "TestSuite"?
```
<include>**/*Test*Suite.java</include>

Comment (chtyim) [inline cdap-examples/deploy_pom.xml]: It is what it was. I didn't make any changes. Just move it to pluginManagement section.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT3625

File: cdap-app-fabric/pom.xml [hunk 2]
```diff
@@ -207,8 +207,9 @@
   <build>
     <plugins>
       <plugin>
-        <groupId>org.scala-tools</groupId>
-        <artifactId>maven-scala-plugin</artifactId>
+        <groupId>net.alchim31.maven</groupId>
```
Comment (bdmogal) [inline cdap-app-fabric/pom.xml]: so this is the new recommended plugin?

File: cdap-archetypes/cdap-spark-scala-archetype/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -91,9 +83,38 @@
...
+          <configuration>
+            <args>
+              <arg>-target:jvm-1.7</arg>
+            </args>
+          </configuration>
+          <executions>
+            <execution>
+              <id>scala-add-source</id>
+              <goals>
+                <goal>add-source</goal>
```
Comment (bdmogal) [inline cdap-archetypes/cdap-spark-scala-archetype/src/main/resources/archetype-resources/pom.xml]: this moves out of the ``process-resources`` phase?

**Comment by bdmogal** [comment]
@chtyim just a couple of questions, since I don't understand this completely. Other than that, LGTM

File: cdap-app-fabric/pom.xml (hunk 2 above)
Comment (chtyim) [inline cdap-app-fabric/pom.xml]: Yes. They are from the same author, just that the old one is no longer maintained.

Comment (bdmogal) [inline cdap-app-fabric/pom.xml]: :+1:

File: cdap-archetypes/cdap-spark-scala-archetype/src/main/resources/archetype-resources/pom.xml (hunk 3 above)
Comment (chtyim) [inline cdap-archetypes/cdap-spark-scala-archetype/src/main/resources/archetype-resources/pom.xml]: By default it happens at initialize phase.. http://davidb.github.io/scala-maven-plugin/add-source-mojo.html

Comment (bdmogal) [inline cdap-archetypes/cdap-spark-scala-archetype/src/main/resources/archetype-resources/pom.xml]: :+1:

**Pull request merged**